src/main/java/model/piece/Cha.java

```diff
@@ -0,0 +1,57 @@
+package model.piece;
+
+import java.util.Map;
+import model.Path;
+import model.Point;
+import model.Team;
+
+public class Cha extends Piece {
+
+    public Cha(Team team) {
+        super(team);
+        pieceName = PieceName.CHA;
+    }
+
+    @Override
+    public boolean isValidPoint(Point beforePoint, Point targetPoint) {
+        return !((beforePoint.x() == targetPoint.x() && beforePoint.y() == targetPoint.y()) || (
+                beforePoint.x() != targetPoint.x() && beforePoint.y() != targetPoint.y()));
+    }
```

Comment:
```suggestion
    @Override
    public boolean isValidPoint(Point beforePoint, Point targetPoint) {
        boolean isStraightMove = beforePoint.x() == targetPoint.x()
                              || beforePoint.y() == targetPoint.y();
        boolean isSamePoint = beforePoint.x() == targetPoint.x()
                           && beforePoint.y() == targetPoint.y();

        return isStraightMove && !isSamePoint;
    }
```

장기, 체스 등 기물을 움직이는 보드게임은 `기물의 움직임을 확인하는 validation` 을 관리하는게 중요하죠.
이는 개발 단계가 아니라 유지보수 하는 단계에서도 중요할거예요.
그러니 복잡하면서도 중요한 로직은 최대한 잘 읽히게 관리해보는건 어떨까요?

Reply:
validation 가독성 리팩터링을 신경쓰지 못했던 것 같습니다. 변경하니 확실히 보기 편하네요. 다른 기물들도 확인해보고 수정하겠습니다!

Reply:
다시보니 `Point`가 `record class` 네요. 
`equals` 메소드를 사용해볼 수도 있을 것 같아요.